packages/router/router.ts

```diff
@@ -814,6 +814,10 @@ export function createRouter(init: RouterInit): Router {
   let unlistenHistory: (() => void) | null = null;
   // Externally-provided functions to call on all state changes
   let subscribers = new Set<RouterSubscriber>();
+  // FIFO queue of previously discovered routes to prevent re-calling on
+  // subsequent navigations to the same path
+  let discoveredRoutesMaxSize = 1000;
+  let discoveredRoutes = new Set<string>();
```

Comment:
Track up to 1000 paths (fifo) internally to avoid re-calling `patchRoutesOnMiss` on subsequent navigations to the same path